node/pkg/txverifier/sui.go

```diff
@@ -164,40 +183,61 @@ func (s *SuiTransferVerifier) processObjectUpdates(objectChanges []ObjectChange,
 	return transferredIntoBridge, numChangesProcessed
 }
 
-func (s *SuiTransferVerifier) ProcessDigest(digest string, suiApiConnection SuiApiInterface, logger *zap.Logger) (uint, error) {
+// ProcessDigestFlagOnly wraps ProcessDigest and only returns true or false, indicating that a specific digest
+// passed validation.
+func (s *SuiTransferVerifier) ProcessDigestFlagOnly(ctx context.Context, digest string, logger *zap.Logger) bool {
+	_, err := s.ProcessDigest(ctx, digest, logger)
+
+	if err != nil {
+		logger.Error("Failed to process digest", zap.Error(err))
+		return false
+	}
+
+	return true
+}
+
+func (s *SuiTransferVerifier) ProcessDigest(ctx context.Context, digest string, logger *zap.Logger) (uint, error) {
+	logger.Debug("Processing digest", zap.String("txDigest", digest))
+
 	// Get the transaction block
-	txBlock, err := suiApiConnection.GetTransactionBlock(digest)
+	txBlock, err := s.suiApiConnection.GetTransactionBlock(ctx, digest)
 
 	if err != nil {
-		logger.Fatal("Error in getting transaction block", zap.Error(err))
+		logger.Error("Error in retrieving transaction block", zap.Error(err))
+		return 0, errors.New("failed to retrieve transaction block")
 	}
 
 	// process all events, indicating funds that are leaving the chain
 	requestedOutOfBridge, numEventsProcessed := s.processEvents(txBlock.Result.Events, logger)
 
+	if numEventsProcessed == 0 {
+		// No valid events were identified, so the digest does not require further processing.
+		return 0, nil
+	}
+
 	// process all object changes, indicating funds that are entering the chain
-	transferredIntoBridge, numChangesProcessed := s.processObjectUpdates(txBlock.Result.ObjectChanges, suiApiConnection, logger)
+	transferredIntoBridge, numChangesProcessed := s.processObjectUpdates(ctx, txBlock.Result.ObjectChanges, logger)
 
-	// TODO: Using `Warn` for testing purposes. Update to Fatal? when ready to go into PR.
-	// TODO: Revisit error handling here.
 	for key, amountOut := range requestedOutOfBridge {
+		keyParts := strings.Split(key, "-")
 
 		if _, exists := transferredIntoBridge[key]; !exists {
-			logger.Warn("transfer-out request for tokens that were never deposited",
-				zap.String("tokenAddress", key))
-			// TODO: Is it better to return or continue here?
+			// This implies that a token leaving the bridge was never deposited into it.
```

Comment:
nit: You could consider writing the error string once here, either as a string or in errors.New, and then log and return that value. That way both will have the same error string and it doesn't need to be duplicated here